modules/ezcontent_preview/src/Plugin/Derivative/DynamicLocalTasks.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Drupal\ezcontent_preview\Plugin\Derivative;
+
+use Drupal\Component\Plugin\Derivative\DeriverBase;
+use Drupal\ezcontent_preview\Utils;
+use Drupal\Core\Url;
+
+/**
+ * Defines dynamic local tasks.
+ */
+class DynamicLocalTasks extends DeriverBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDerivativeDefinitions($base_plugin_definition) {
+    $decoupledRoutes = \Drupal::entityTypeManager()->getStorage('ezcontent_preview')->loadMultiple();
+    if ($decoupledRoutes) {
+      $current_path = \Drupal::service('path.current')->getPath();
+      $path_args = explode('/', $current_path);
+      if ($path_args && $path_args[1] == 'node') {
+        $node = \Drupal\node\Entity\Node::load($path_args[2]);
```

Comment:
Lets inject this object

Reply:
not possible I believe as on dynamic tabs not getting any DI feature.